src/Module.zig

```diff
@@ -3514,6 +3528,587 @@ fn semaFile(mod: *Module, file: *File) SemaError!void {
+                        .false => .bool_false,
+                        .null => .null_value,
+                        .nan => self.mod.intern(.{ .float = .{
+                            .ty = try self.mod.intern(.{ .simple_type = .comptime_float }),
+                            .storage = .{ .f128 = std.math.nan(f128) },
+                        } }),
+                        .inf => try self.mod.intern(.{ .float = .{
+                            .ty = try self.mod.intern(.{ .simple_type = .comptime_float }),
+                            .storage = .{ .f128 = std.math.inf(f128) },
+                        } }),
+                    };
+                }
+                return self.fail(.{ .node_abs = node }, "use of unknown identifier '{s}'", .{ litIdent.bytes });
+            },
+            .number_literal, .char_literal => return self.number(node, null),
+            .negation => return self.number(data[node].lhs, node),
+            .enum_literal => return self.mod.intern_pool.get(gpa, .{
+                .enum_literal = try self.identAsNullTerminatedString(main_tokens[node]),
+            }),
+            .string_literal => {
+                const token = main_tokens[node];
+                const raw_string = self.file.tree.tokenSlice(token);
+
+                var bytes = std.ArrayListUnmanaged(u8){};
+                defer bytes.deinit(gpa);
+
+                switch (try std.zig.string_literal.parseWrite(bytes.writer(gpa), raw_string)) {
+                    .success => {},
+                    .failure => |err| {
+                        const offset = self.file.tree.tokens.items(.start)[token];
+                        return AstGen.failWithStrLitError(
+                            self,
+                            failWithStrLitError,
+                            err,
+                            token,
+                            raw_string,
+                            offset,
+                        );
+                    }
+                }
+
+                const array_ty = try self.mod.arrayType(.{
+                    .len = bytes.items.len,
+                    .sentinel = .zero_u8,
+                    .child = .u8_type,
+                });
+                const val = try self.mod.intern(.{ .aggregate = .{
+                    .ty = array_ty.toIntern(),
+                    .storage = .{ .bytes = try self.mod.intern_pool.getOrPutString(gpa, bytes.items, .maybe_embedded_nulls) },
+                } });
+                const ptr_ty = (try self.mod.ptrType(.{
+                    .child = array_ty.toIntern(),
+                    .flags = .{
+                        .alignment = .none,
+                        .is_const = true,
+                        .address_space = .generic,
+                    },
+                })).toIntern();
+                return try self.mod.intern(.{ .ptr = .{
+                    .ty = ptr_ty,
+                    .base_addr = .{ .anon_decl = .{ .val = val, .orig_ty = ptr_ty } },
```

Comment:
I use `anon_decl` in a few places to intern slices. I'm not 100% sure if this is the correct way to do this or not.

Reply:
Yup, `.anon_decl` is the correct thing to use to represent a pointer to a constant comptime-known value.